op-chain-ops/cmd/celo-migrate/README.md

```diff
@@ -0,0 +1,87 @@
+
+### Block migration
+
+The block migration itself has two parts: It first migrates the ancient / frozen blocks, which is all blocks before the last 90000. Because the ancients db is append-only, it copies these blocks into a new database after making the necessary transformations. The script then copies the rest of the chaindata directory (excluding `/ancients`) using the system level `rsync` command. All non-ancient blocks are then transformed in-place in the new db, leaving the old db unchanged.
+
+### State migration
+
+After all blocks have been migrated, the script performs a series of modifications to the state db. This is also done in-place in the `--new-db` directory. First, the state migration deploys the L2 smart contracts by iterating through the genesis allocs passed to the script and setting the nonce, balance, code and storage for each address accordingly, overwritting existing data if necessary. For testnets, the state migration script will also migrate core celo contracts to their mainnet addresses for consistency across celo networks. Finally, the state migration will commit the state changes to produce a new state root and create the first Cel2 block.
+
+### Notes
+
+Once the state changes are complete the migration is finished. The longest running section of the script is the ancients migration, and it can be resumed / skipped if interupted part way. The rest of the script cannot be resumed and will restart from the last migrated ancient block if interupted or re-run.
+
+The script outputs a `rollup-config.json` file that is passed to the sequencer in order to start the L2 network.
+
+See `--help` for how to run each portion of the script individually, along with other configuration options.
+
+### Running the script
+
+First, build the script by running
+
+```bash
+make celo-migrate
+```
+
+from the `op-chain-ops` directory.
+
+You can then run the script as follows.
+
+```bash
+go run ./cmd/celo-migrate --help
+```
+
+NOTE: You will need `rsync` to run this script if it's not already installed
+
+#### Running with local test setup (Alfajores / Holesky)
+
+To test the script locally, we can migrate an alfajores database and use Holesky as our L1. The input files needed for this can be found in `./testdata`. The necessary smart contracts have already been deployed on Holesky.
+
+Pull down the latest alfajores database snapshot
+
+```bash
+gcloud alpha storage cp gs://celo-chain-backup/alfajores/chaindata-latest.tar.zst alfajores.tar.zst
+```
+
+Unzip and rename
+
+```bash
+tar --use-compress-program=unzstd -xvf alfajores.tar.zst
+mv chaindata ./data/alfajores_old
+```
+
+Run script with test configuration
+
+```bash
+go run ./cmd/celo-migrate full \
+--deploy-config ./cmd/celo-migrate/testdata/deploy-config-holesky-alfajores.json \
+--l1-deployments ./cmd/celo-migrate/testdata/deployment-l1-holesky.json \
+--l1-rpc https://ethereum-holesky-rpc.publicnode.com  \
+--l2-allocs ./cmd/celo-migrate/testdata/alloc-alfajores.json \
```

Comment:
Probably we shouldn't include the alloc-alfajores.json as it is huge and it can be generated.

Reply:
What do you mean we shouldn't include it? Like we should generate it as part of the script?

Reply:
No I meant include as a file in git. It's fine as a parameter but it is the output of a forge script, so its better to let people generate it rather than committing it to the repo.

Reply:
We also need to explain to users (probably in the flag description) where they get the allocs from?

Reply:
I don't think we need to explain that in the flag's description, it's abit involved and should be enough to have it in a guide.

Reply:
I think at least we need to say something, otherwise users have no hint of what the allocs are or where they should get them.